logging/include/cocaine/idl/logging.hpp

```diff
@@ -0,0 +1,228 @@
+            return "get";
+        }
+
+        typedef boost::mpl::list<
+            // Name used for log routing and filtering.
+            std::string
+        >::type argument_type;
+
+        typedef named_log_tag dispatch_type;
+    };
+
+    struct list_loggers {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "list_loggers";
+        }
+
+        typedef option_of<
+            std::vector<std::string>
+        >::tag upstream_type;
+    };
+
+    struct set_filter {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "set_filter";
+        }
+
+        typedef boost::mpl::list<
+            // Logger name
+            std::string,
+            // Filter itself
+            logging::filter_t,
+            // Time to store filter
+            uint64_t
+        >::type argument_type;
+
+        typedef option_of<
+        // Id of the filter created.
+            logging::filter_t::id_type
+        >::tag upstream_type;
+    };
+
+    struct remove_filter {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "remove_filter";
+        }
+
+        typedef boost::mpl::list<
+            logging::filter_t::id_type
+        >::type argument_type;
+
+        // true if filter was deleted,
+        // false if it didn't exist,
+        // error in other cases
+        typedef option_of<
+            bool
+        >::tag upstream_type;
+    };
+
+    struct list_filters {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "list_filters";
+        }
+
+        typedef option_of<
+            std::vector<std::tuple<
+                std::string,
+                logging::filter_t::representation_t,
+                logging::filter_t::id_type,
+                logging::filter_t::disposition_t
+            >>
+        >::tag upstream_type;
+    };
+
+
+    struct set_cluster_filter {
+        typedef base_log_tag tag;
+
+        static const char* alias() {
+            return "set_cluster_filter";
+        }
+
+        typedef boost::mpl::list<
+            // Logger name
+            std::string,
+            // Filter itself
+            logging::filter_t,
+            // Time to store filter
+            uint64_t
+        >::type argument_type;
+
+        typedef option_of<
+            // Id of the filter created.
+            logging::filter_t::id_type
+        >::tag upstream_type;
+    };
+};
+
+struct named_log {
+    struct emit {
+        typedef named_log_tag tag;
+
+        static const char* alias() {
+            return "emit";
+        }
+
+        typedef boost::mpl::list<
+         /* Log message. Some meaningful string, with no explicit limits on its length, although
+            underlying loggers might silently truncate it. */
+            std::string,
+        /* Log severity*/
+            optional_with_default<unsigned int, 1u>,
```

Comment:
First I was thinking about making severity just another optional attribute, but it's required by logger in new BH. Ok, I'll make it mandatory.